cpp/ql/src/Security/CWE/CWE-190/ArithmeticUncontrolled.ql

```diff
@@ -18,8 +18,16 @@ import semmle.code.cpp.security.TaintTracking
 import semmle.code.cpp.rangeanalysis.SimpleRangeAnalysis
 import TaintedWithPath
 
+string getAMinPattern() { result = ["%min%", "l%"] }
+
+string getAMaxPattern() { result = ["%max%", "%bound%", "h%"] }
+
 predicate isUnboundedRandCall(FunctionCall fc) {
-  fc.getTarget().getName() = "rand" and not bounded(fc)
+  exists(Function func | func = fc.getTarget() |
+    func.getName() = "rand" and
+    not bounded(fc) and
+    not func.getAParameter().getName().toLowerCase().matches([getAMinPattern(), getAMaxPattern()])
+  )
 }
```

Comment:
I presume the original code is trying to match only the `int rand(void)` function from `stdlib.h`, not any other function named `rand`. The `rand` function inadvertently matched in `gemrb/gemrb` is a member function of a class. Just switching from `getName` to `hasGlobalOrStdOrBslName` should solve that problem. If you want to further restrict it, you can check that the matched function has no parameters.

Reply:
I thought it was actually a feature of the query that we picked any function named `rand` as a possible source of randomness. But looking at the [results](https://lgtm.com/rules/2156240609/alerts/) I see that it's pretty much always a FP whenever it's not `std::rand`. So I'll change the query to use `hasGlobalOrStdOrBslName` instead.

Reply:
Fixed in 615c805b2ce0395b9761e4ca03e472f59c999abe.